src/vanilla/TransformerTS.cs

```diff
@@ -33,9 +33,61 @@ public override CodeModelTS TransformCodeModel(CodeModel cm)
             EnsureParameterMethodSet(codeModel);
             CreateUniqueParameterMapperNames(codeModel);
 
+            if (codeModel.ModelDateAsString)
+            {
+                ConvertDateToString(codeModel);
+            }
+
             return codeModel;
         }
 
+        private void ConvertDateToString(CodeModelTS codeModel)
```

Comment:
Unfortunately this proved the most expedient approach, as passing the CodeModel down through all the TSType, TSProperty etc methods from templates was unreasonably difficult (`IModelType.CodeModel` is not populated and can't be generically populated because there's no setter on the interface)